docs/hugo/content/design/ADR-2023-07-Goal-Seeking-KeyVaults/_index.md

```diff
@@ -0,0 +1,154 @@
+
+Currently, the resource will end up in an unready state that ASO cannot automatically resolve. When ASO attempts to reconcile the resource, the reconciliation will fail with an error that the requested KeyVault name is unavailable, and the resource will remain in an unready state.
+
+Ideally, ASO would be able to automatically take the appropriate remedial action, restoring the functionality of the application running on the cluster.
+
+#### Staging Deployment
+
+It's typical for an application in development to be deployed into a staging environment for testing. In order to test the deployment process, this is often done by completely deleting the prior release and deploying everything from scratch.
+
+If cleanup of the prior release into Staging has soft-deleted the KeyVault, the new deployment will fail with an error that the requested KeyVault name is unavailable.
+
+As a consequence, users currently need to separately purge or recover the KeyVault before the new deployment will successfully provision the resources required by the application. They may do this manually, or by running a separate script.
+
+Ideally, staging deployments driven by ASO would work end to end without additional intervention.
+
+### Current Properties
+
+The [KeyVault](https://azure.github.io/azure-service-operator/reference/keyvault/v1api20210401preview/#keyvault.azure.com/v1api20210401preview.VaultProperties) resource has the following properties relevant to this discussion.
+
+#### createMode
+
+The vault’s create mode to indicate whether the vault need to be recovered or not.
+
+If set to `default` (or if not specified), KeyVault creation will only succeed if there is no existing soft-deleted KeyVault with the same name. If there is such a soft-deleted KeyVault, creation of the KeyVault will fail.
+
+If set to `recover`, KeyVault creation will only succeed if there is an existing soft-deleted KeyVault with the same name. If there is no such KeyVault, creation of the KeyVault will fail.
+
+There is no `purge` option, so it's not possible to purge a KeyVault and create a replacement in a single operation. We can reasonably infer that the Product Group did this deliberately, to prevent accidental purging of a KeyVault.
+
+#### enablePurgeProtection
+
+Specify whether protection against purge is enabled for this vault. 
+
+Setting this property to **true** activates protection against purge for this vault and its content - only the Key Vault service may initiate a hard, irrecoverable deletion. The setting is effective only if soft delete is also enabled. Enabling this functionality is irreversible - that is, the property does not accept **false** as its value.
+  
+#### enableSoftDelete
+
+Specify whether the ‘soft delete’ functionality is enabled for this key vault. 
+
+If it’s not set to any value (**true** or **false**) when creating new key vault, it will be set to true by default. Once set to true, it cannot be reverted to false.
```

Comment:
Also mention about `soft-delete` is deprecated and will be enforced in the future AKV releases

Reply:
Done.